test/upgrade/installation/serverless.go

```diff
@@ -14,14 +15,32 @@ import (
 	"k8s.io/client-go/util/retry"
 )
 
-func UpgradeServerlessTo(ctx *test.Context, csv, source string) error {
+const (
+	DefaultInstallPlanTimeout = 15 * time.Minute
+)
+
+func UpgradeServerlessTo(ctx *test.Context, csv, source string, timeout time.Duration) error {
 	if _, err := test.UpdateSubscriptionChannelSource(ctx, test.Flags.Subscription, test.Flags.UpgradeChannel, source); err != nil {
 		return err
 	}
 
-	installPlan, err := test.WaitForInstallPlan(ctx, test.OperatorsNamespace, csv, source)
+	installPlan, err := test.WaitForInstallPlan(ctx, test.OperatorsNamespace, csv, source, timeout)
 	if err != nil {
-		return err
+		if !strings.Contains(err.Error(), "not found") {
+			return err
+		}
```

Comment:
nit: would the `!apierrors.IsNotFound(err)` work here?

Reply:
No. It can't be used because the actual error is this, without any additional status:
```
var ErrWaitTimeout = errors.New("timed out waiting for the condition")
```
However, I could change the condition to this:
```
if !errors.Is(err, wait.ErrWaitTimeout) {
	return err
}
```